### PR TITLE
Support invoke super on more method types

### PR DESCRIPTION
**Pull request by jimmyhmiller**

### CODE REVIEW

File: yjit/src/codegen.rs [hunk 1]
```diff
@@ -5906,6 +5925,103 @@ fn gen_invokesuper(
         VM_METHOD_TYPE_CFUNC => {
             gen_send_cfunc(jit, ctx, asm, ocb, ci, cme, block, ptr::null(), ci_flags, argc)
         }
+        VM_METHOD_TYPE_IVAR => {
+
+            let recv_idx = argc;
+
+            let comptime_recv = jit_peek_at_stack(jit, ctx, recv_idx as isize);
+            let comptime_recv_klass = comptime_recv.class_of();
+
```
Comment (maximecb) [inline yjit/src/codegen.rs:5934]:
```suggestion
```

File: yjit/src/codegen.rs [hunk 2]
```diff
@@ -5906,6 +5925,103 @@ fn gen_invokesuper(
         VM_METHOD_TYPE_CFUNC => {
             gen_send_cfunc(jit, ctx, asm, ocb, ci, cme, block, ptr::null(), ci_flags, argc)
         }
+        VM_METHOD_TYPE_IVAR => {
+
```
Comment (maximecb) [inline yjit/src/codegen.rs:5929]:
```suggestion
```

**Review by maximecb — COMMENTED** [review]
Code looks good, but could you include the railsbench stats before/after ?

File: yjit/src/codegen.rs [hunk 3]
```diff
@@ -5831,20 +5833,36 @@ fn gen_invokesuper(
...
 
     // Do method lookup
-    let cme = unsafe { rb_callable_method_entry(comptime_superclass, mid) };
+    let mut cme = unsafe { rb_callable_method_entry(comptime_superclass, mid) };
 
     if cme.is_null() {
         return CantCompile;
     }
 
+
```
Comment (noahgibbs) [inline yjit/src/codegen.rs:5847]: Similarly, probably don't need these 3 extra lines of whitespace.

File: yjit/src/codegen.rs [hunk 4]
```diff
@@ -5906,6 +5925,103 @@ fn gen_invokesuper(
...
+                return CantCompile;
+            } else if ci_flags & VM_CALL_ARGS_BLOCKARG != 0 {
+                gen_counter_incr!(asm, send_block_arg);
+                return CantCompile;
+            } else {
+                let ivar_name = unsafe { get_cme_def_body_attr_id(cme) };
+                return gen_set_ivar(jit, ctx, asm, comptime_recv, ivar_name, ci_flags, argc);
+            }
+        }
+
```
Comment (XrXr) [inline yjit/src/codegen.rs:6024]:
```suggestion
```

**Comment by maximecb** [comment]
> Do you know if we already have a test that hits using an alias via super()?

👍 always good to be extra super diligent if we don't have such a test.

**Review by XrXr — CHANGES_REQUESTED** [review]
There is a behavior difference:
```ruby
# review for 5516bf4580da2b692d68b10b3734f04e3adeba7a Support invoke super on more method types

class Parent
  attr_accessor :foo
end

class Splat < Parent
  def foo=(arg)
    super(*arg)
  end
end

s = Splat.new
s.foo = [100]
p s.foo # 100 in interpreter, [100] with YJIT
```
You need to replicate this check I think:
https://github.com/ruby/ruby/blob/ba3b40a9aeb752d1b3e4c87748c977bfd3cf6f2c/yjit/src/codegen.rs#L5300-L5303
Since we're close to the release I'm a bit nervous about other things I might've missed, but maybe it's fine.

**Comment by jimmyhmiller** [comment]
Closing since we are close to release Can revisit some other time.

**Pull request closed without merging**